contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -3329,6 +3335,45 @@ static void bbf_ProcessUtility(PlannedStmt *pstmt,
 				queryEnv, dest, qc);
 }
 
+/*
+ * Update the pg_type catalog entry for the given name to have
+ * typbyval set to the given value.
+ */
+static void set_pgtype_byval(List *name, bool byval)
+{
+	Relation	catalog;
+	TypeName   *typename;
+	HeapTuple	tup;
+	
+	Datum		values[Natts_pg_type];
+	bool		nulls[Natts_pg_type];
+	bool		replaces[Natts_pg_type];
+	HeapTuple	newtup;
+
+	/*
+		* Table types need to set the typbyval column in pg_type to 't'
+		*/
+	catalog = table_open(TypeRelationId, RowExclusiveLock);
+	typename = makeTypeNameFromNameList(name);
+	tup = typenameType(NULL, typename, NULL);
+
+	/* Update the current type's tuple */
+	memset(values, 0, sizeof(values));
+	memset(nulls, 0, sizeof(nulls));
+	memset(replaces, 0, sizeof(replaces));
+	replaces[Anum_pg_type_typbyval - 1] = true;
+	values[Anum_pg_type_typbyval - 1] = BoolGetDatum(byval);
+
+	newtup = heap_modify_tuple(tup, RelationGetDescr(catalog), values, nulls, replaces);
+	CatalogTupleUpdate(catalog, &newtup->t_self, newtup);
+
+	/* Clean up */
```

Comment:
what about `newtup`?

Reply:
the other code examples I borrowed this implementation from do not clean newtup, so I am not sure if it is necessary/safe to manually do so in here.